core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -150,6 +150,20 @@ public void initialize(String name, Map<String, String> properties) {
       Thread.currentThread().interrupt();
       throw new UncheckedInterruptedException(e, "Interrupted in call to initialize");
     }
+
+    try {
+      updateSchemaIfRequired();
```

Comment:
Ryan mentioned in https://github.com/apache/iceberg/pull/9765#discussion_r1496224382 that the error handling should move inside `updateSchemaIfRequired()` and we should probably do the same for `initializeCatalogTables()`. Otherwise the `initialize()` method gets quite bloated

Reply:
>  I think the try/catch should move into the `initializeCatalogTables` method and `updateSchemaIfRequired` should have its own error handling.

Based on this it's what I mentioned previously: the error handling should move *into* `initializeCatalogTables` + `updateSchemaIfRequired` should have its own (and in that case I would also move it into the method itself)

Reply:
Catcha, let me move the error handling in the methods.